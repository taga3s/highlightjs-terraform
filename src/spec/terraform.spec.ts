import fs from "node:fs";
import path from "node:path";
import hljs from "highlight.js";
import { describe } from "vitest";
import { it } from "vitest";
import { expect } from "vitest";
import { definer } from "../index.ts";

hljs.registerLanguage("terraform", definer);

describe("highlight bundle", () => {
	it("highlights terraform (base-input)", () => {
		const input = fs.readFileSync(
			path.resolve(__dirname, "./base-input.txt"),
			"utf-8",
		);

		const { value: result, language } = hljs.highlightAuto(input, [
			"terraform",
		]);

		expect(language).toBe("terraform");
		expect(result).toMatchSnapshot();
	});

	it("highlights terraform (base2-input)", () => {
		const input = fs.readFileSync(
			path.resolve(__dirname, "./base2-input.txt"),
			"utf-8",
		);

		const { value: result, language } = hljs.highlightAuto(input, [
			"terraform",
		]);

		expect(language).toBe("terraform");
		expect(result).toMatchSnapshot();
	});

	it("highlights terraform (base3-input)", () => {
		const input = fs.readFileSync(
			path.resolve(__dirname, "./base3-input.txt"),
			"utf-8",
		);

		const { value: result, language } = hljs.highlightAuto(input, [
			"terraform",
		]);

		expect(language).toBe("terraform");
		expect(result).toMatchSnapshot();
	});
});
