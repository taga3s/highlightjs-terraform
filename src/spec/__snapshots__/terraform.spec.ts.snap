// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`highlight bundle > highlights terraform (base-input) 1`] = `
"<span class="hljs-comment"># Terraform (hcl)</span>
<span class="hljs-keyword">
variable</span> <span class="hljs-string">&quot;string&quot;</span> {
  default = <span class="hljs-string">&quot;string_test&quot;</span> <span class="hljs-comment"># test inline comment</span>
}
<span class="hljs-keyword">
resource</span> <span class="hljs-string">&quot;azurerm_subnet&quot;</span> <span class="hljs-string">&quot;sn1&quot;</span> {
  count                = <span class="hljs-number">1</span>
  number               = <span class="hljs-number">2.3</span>
  boolean              = <span class="hljs-literal">true</span>
  type                 = <span class="hljs-string">&quot;<span class="hljs-subst">\${var.string}</span>&quot;</span>
  name                 = <span class="hljs-string">&quot;<span class="hljs-subst">\${var.prefix}</span>-sn&quot;</span>
  resource_group_name  = <span class="hljs-string">&quot;<span class="hljs-subst">\${azurerm_resource_group.rg1.name}</span>&quot;</span>
  virtual_network      = <span class="hljs-string">&quot;<span class="hljs-subst">\${<span class="hljs-meta">cidrsubnet(<span class="hljs-string">&quot;test&quot;</span>,<span class="hljs-number">3</span>,<span class="hljs-meta">test()</span>)</span>}</span>&quot;</span>
  address_prefix       = <span class="hljs-string">&quot;<span class="hljs-subst">\${<span class="hljs-meta">cidrsubnet(<span class="hljs-string">&quot;<span class="hljs-subst">\${var.vnets[<span class="hljs-string">&quot;<span class="hljs-subst">\${var.location}</span>&quot;</span>,<span class="hljs-string">&quot;test1&quot;</span>]}</span>&quot;</span>,<span class="hljs-string">&quot;<span class="hljs-subst">\${var.network_size}</span>&quot;</span>,<span class="hljs-number">3</span>,<span class="hljs-string">&quot;test&quot;</span>)</span>}</span>&quot;</span>

  tags = <span class="hljs-string">&quot;<span class="hljs-subst">\${<span class="hljs-meta">merge(local.tags, <span class="hljs-meta">map(<span class="hljs-string">&quot;update&quot;</span>,<span class="hljs-string">&quot;<span class="hljs-subst">\${<span class="hljs-meta">timestamp()</span>}</span>&quot;</span>)</span>)</span>}</span>&quot;</span>
}
<span class="hljs-keyword">
output</span> <span class="hljs-string">&quot;public_ip_address&quot;</span> {
  value = [<span class="hljs-string">&quot;<span class="hljs-subst">\${azurerm_public_ip.pip1.*.ip_address}</span>&quot;</span>]
}
<span class="hljs-keyword">

terraform</span> {
<span class="hljs-keyword">  backend</span> <span class="hljs-string">&quot;azure&quot;</span> {
    storage_account_name = <span class="hljs-string">&quot;azure&quot;</span>
  }
}
"
`;
